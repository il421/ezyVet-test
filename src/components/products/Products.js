import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addItemToCart } from '../../actions/cart.js';

import '../../styles/products/products.css';

import Product from './Product.js';

export class Products extends Component {

  products = [
    {
      id: 1,
      name: 'Sledgehammer',
      price: 125.75,
      mediaUrl: 'http://placehold.it/200x200'
    },
    {
      id: 2,
      name: 'Axe',
      price: 190.50,
      mediaUrl: 'http://placehold.it/200x200'
    },
    {
      id: 3,
      name: 'Bandsaw',
      price: 562.13,
      mediaUrl: 'http://placehold.it/200x200'
    },{
      id: 4,
      name: 'Chisel',
      price: 12.9,
      mediaUrl: 'http://placehold.it/200x200'
    },
    {
      id: 5,
      name: 'Hacksaw',
      price: 18.45,
      mediaUrl: 'http://placehold.it/200x200'
    }
  ];

  render() {
    return (
      <section className="products">
        {
          this.products.map((product) => (
            <Product
              key={ product.id }
              { ...product }
              addToCart={ this.props.addItemToCart }
            />
          ))
        }
      </section>
    );
  }
}

const mapDispatchToProps = (dispatch) => (
  {
    addItemToCart: ({id, name, price, qty}) => dispatch(addItemToCart({id, name, price, qty}))
  }
);

export default connect(undefined, mapDispatchToProps)(Products);
