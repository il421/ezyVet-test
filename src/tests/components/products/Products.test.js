import React from 'react';
import { shallow } from 'enzyme';
import { Products } from '../../../components/products/Products.js';

test('should render Products without crashing', () => {
  const wrapper = shallow(<Products />);
  expect(wrapper).toMatchSnapshot();
});
