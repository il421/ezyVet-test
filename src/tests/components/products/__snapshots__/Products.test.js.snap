// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Products without crashing 1`] = `
<section
  className="products"
>
  <Product
    id={1}
    key="1"
    mediaUrl="http://placehold.it/200x200"
    name="Sledgehammer"
    price={125.75}
  />
  <Product
    id={2}
    key="2"
    mediaUrl="http://placehold.it/200x200"
    name="Axe"
    price={190.5}
  />
  <Product
    id={3}
    key="3"
    mediaUrl="http://placehold.it/200x200"
    name="Bandsaw"
    price={562.13}
  />
  <Product
    id={4}
    key="4"
    mediaUrl="http://placehold.it/200x200"
    name="Chisel"
    price={12.9}
  />
  <Product
    id={5}
    key="5"
    mediaUrl="http://placehold.it/200x200"
    name="Hacksaw"
    price={18.45}
  />
</section>
`;
